### PR TITLE
DCS-763 restructuring routes and getting rid of intermediate routers

### PR DESCRIPTION
**Pull request by andrewrlee**
Previously had a folder structure like:

```
/controllers
 - controller1
 - controller2
 ... 
 - controller20

/routers
 - router def for controller 1
 - router def for controller 2
 ...
```

now restructuring to:
```
/routes
  /createBooking
    - controller1
    - controller2
    -  index.ts (exposing router for creating booking)
  /requestBooking
    - controller3
    - controller4
    -  index.ts (exposing router for request booking)
   ...
```

Hoping the new structure will give space for creation of utility modules to allow reduction of the size of some of the routes

### CODE REVIEW

File: backend/routes/createBooking/selectCourtAppointmentRooms.js [hunk 1]
```diff
@@ -83,13 +83,13 @@ const validate = ({
   return errors
```
Comment (andrewrlee) [inline backend/routes/createBooking/selectCourtAppointmentRooms.js:83]: I want to rename some of this - alot of the names are mostly implicit:

selectCourtAppointmentRooms -> selectRooms

File: backend/routes/index.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import express, { Router } from 'express'
+
+import { Services } from '../services'
+
+import requestBookingRoutes from './requestBooking'
+import createBookingRoutes from './createBooking'
+import changeBookingRoutes from './changeBooking'
+import viewBookingsRoutes from './viewBookings'
+
+const router = express.Router()
```
Comment (andrewrlee) [inline backend/routes/index.ts:10]: start point for new routes

File: backend/tests/deleteBooking.test.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { Request, Response } from 'express'
```
Comment (andrewrlee) [inline backend/tests/deleteBooking.test.ts:1]: would also like to not store the tests in a separate folder (a lot of tests are alongside the code)

File: backend/routes/createBooking/index.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  )
+
+  router.get(
+    '/offenders/:offenderNo/confirm-appointment',
+    withRetryLink('/prisoner-search'),
+    asyncMiddleware(confirmAppointment(services))
+  )
+
+  {
+    const { index, validateInput, goToCourtSelection } = addCourtAppointment(services.prisonApi)
```
Comment (andrewrlee) [inline backend/routes/createBooking/index.ts:29]: this whole file is a bit gnarly - Got a nice tidy up as a follow up.

Comment (andrewrlee) [reply backend/routes/createBooking/index.ts:29]: sneaky peek 🎅 : 

```js
import express, { Router } from 'express'

import confirm from './confirmController'
import startController from './startController'
import selectCourt from './selectCourtController'
import selectRooms from './selectRoomsController'
import prisonerSearch from './prisonerSearchController'

import withRetryLink from '../../middleware/withRetryLink'
import asyncMiddleware from '../../middleware/asyncMiddleware'
import checkAppointmentRooms from '../../middleware/checkAppointmentRooms'

import { Services } from '../../services'

export default function createRoutes(services: Services): Router {
  const router = express.Router({ mergeParams: true })

  const checkRooms = asyncMiddleware(checkAppointmentRooms(services))

  router.get('/prisoner-search', withRetryLink('/'), asyncMiddleware(prisonerSearch(services)))

  {
    const { index, validateInput, goToCourtSelection } = startController(services)
    const path = '/:agencyId/offenders/:offenderNo/add-court-appointment'
    router.get(path, withRetryLink('/prisoner-search'), asyncMiddleware(index))
    router.post(path, validateInput, checkRooms, goToCourtSelection)
  }

  {
    const { index, post } = selectCourt(services)
    const path = '/:agencyId/offenders/:offenderNo/add-court-appointment/select-court'
    router.get(path, asyncMiddleware(index))
    router.post(path, asyncMiddleware(post))
  }

  {
    const { index, validateInput, createAppointments } = selectRooms(services)
    const path = '/:agencyId/offenders/:offenderNo/add-court-appointment/select-rooms'
    router.get(path, asyncMiddleware(index))
    router.post(path, validateInput, checkRooms, asyncMiddleware(createAppointments))
  }

  router.get(
    '/offenders/:offenderNo/confirm-appointment',
    withRetryLink('/prisoner-search'),
    asyncMiddleware(confirm(services))
  )

  return router
}



```

**Review by bell-pa — APPROVED** [review]
LGTM

**Pull request merged**